src/geo/transform.js

```diff
@@ -278,50 +279,26 @@ class Transform {
     scaleZoom(scale: number) { return Math.log(scale) / Math.LN2; }
 
     project(lnglat: LngLat) {
-        return new Point(
-            this.lngX(lnglat.lng),
-            this.latY(lnglat.lat));
+        const lat = clamp(lnglat.lat, -this.maxValidLatitude, this.maxValidLatitude);
+        const clamped = new LngLat(lnglat.lng, lat);
+        const coord = MercatorCoordinate.fromLngLat(clamped);
+        return new Point(coord.x * this.worldSize, coord.y * this.worldSize);
```

Comment:
My only concern with this change is that we significantly increase the number of allocations in potentially hot code, like here — instead of a single `Point` allocation, we do 3 — `Point`, `MercatorCoordinate`, `Point` again.

Reply:
Do you think this is significant enough to warrant splitting out the projection into `lngToMercatorX(...)` and `latToMercatorY(...)` for internal use?

Reply:
Does 0d4c2ca look good?